### PR TITLE
Add the clone feature back to each figure

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses:
- Task: https://github.com/idmc-labs/helix2.0-meta/issues/198

## Changes:
- Change the functionality of "Add Figure" button
- Add "Clone Figure" button to each figure

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: src/views/Entry/EntryForm/index.tsx [hunk 1]
```diff
@@ -688,6 +654,30 @@ function EntryForm(props: EntryFormProps) {
         [onValueChange, value, notify],
     );
 
+    const handleFigureClone = useCallback(
+        (index: number) => {
```
Comment (tnagorra) [inline src/views/Entry/EntryForm/index.tsx:658]: Lets' pass the value instead of index. So that we do not need to iterate on the figures list

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/Entry/EntryForm/index.tsx [hunk 2]
```diff
@@ -688,6 +654,31 @@ function EntryForm(props: EntryFormProps) {
         [onValueChange, value, notify],
     );
 
+    const handleFigureClone = useCallback(
+        (key: string) => {
+            const oldFigure = value.figures?.find((fig) => fig.uuid === key);
+            if (!oldFigure) {
+                return;
+            }
+            console.log('Obtained key::>>', key);
```
Comment (tnagorra) [inline src/views/Entry/EntryForm/index.tsx:663]: Let's remove console.log meant for debugging

File: src/views/Entry/EntryForm/FigureInput/index.tsx [hunk 3]
```diff
@@ -898,6 +902,13 @@ function FigureInput(props: FigureInputProps) {
                 subSection
                 actions={editMode && (
                     <>
+                        <Button
+                            name={figureKey}
```
Comment (tnagorra) [inline src/views/Entry/EntryForm/FigureInput/index.tsx:906]:
```suggestion
                            name={value}
```

File: src/views/Entry/EntryForm/FigureInput/index.tsx [hunk 4]
```diff
@@ -906,7 +917,7 @@ function FigureInput(props: FigureInputProps) {
                             Clear
                         </Button>
                         <Button
-                            name={index}
+                            name={+figureKey}
```
Comment (tnagorra) [inline src/views/Entry/EntryForm/FigureInput/index.tsx:920]:
```suggestion
                            name={index}
```

File: src/views/Entry/EntryForm/index.tsx [hunk 5]
```diff
@@ -688,6 +654,31 @@ function EntryForm(props: EntryFormProps) {
         [onValueChange, value, notify],
     );
 
+    const handleFigureClone = useCallback(
+        (key: string) => {
```
Comment (tnagorra) [inline src/views/Entry/EntryForm/index.tsx:658]: Instead of reading key from handleFigureClone, directly get the value from the component

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/Entry/EntryForm/FigureInput/index.tsx (hunk 3 above)
Comment (priyesh777) [reply src/views/Entry/EntryForm/FigureInput/index.tsx:906]: Not able to pass an object or an array through "name" field

Comment (tnagorra) [reply src/views/Entry/EntryForm/FigureInput/index.tsx:906]: In that case, we can leave the name prop `undefined` and pass a custom function for prop `onClick`

Comment (priyesh777) [reply src/views/Entry/EntryForm/FigureInput/index.tsx]: This has been implemented

**Pull request merged**